source/extensions/filters/network/postgres_proxy/postgres_decoder.h

```diff
@@ -87,20 +85,23 @@ class DecoderImpl : public Decoder, Logger::Loggable<Logger::Id::filter> {
   // when a specific message has been decoded.
   using MsgAction = std::function<void(DecoderImpl*)>;
 
-  // MsgProcessor has two fields:
+  // MessageProcessor has the following fields:
   // first - string with message description
-  // second - vector of Decoder's methods which are invoked when the message
+  // second - function which instantiates a Message object of specific type
+  // which is capable of parsing the message's body.
+  // third - vector of Decoder's methods which are invoked when the message
   // is processed.
-  using MsgProcessor = std::pair<std::string, std::vector<MsgAction>>;
+  using MessageProcessor =
+      std::tuple<std::string, std::function<std::unique_ptr<Message>()>, std::vector<MsgAction>>;
```

Comment:
can you make a separate `using` nickname for the function type, and then also use that name rather than `auto` in `genDebugMessage()` and possibly other places? I think that would make it easier to understand when reading.

Reply:
Added `MsgBodyReader` type.